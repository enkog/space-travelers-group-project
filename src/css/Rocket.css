.rocket-card {
  margin: 2rem 2rem;
}

.rocket-image {
  width: 15rem;
  margin: 0 2rem;
}
