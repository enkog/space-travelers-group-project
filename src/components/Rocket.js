import PropTypes from 'prop-types';
import { Badge } from 'react-bootstrap';
import ReservationRocket from './ReservationRocket';
import '../css/Rocket.css';

const Rocket = ({
  id, name, type, image,
}) => (
  <div className="rocket-card d-flex">
    <img src={image} alt="Rocket" className="rocket-image" />
    <div>
      <h1 className="rocket-name">{name}</h1>
      <p className="rocket-type">
        <Badge className="tag">Reserved</Badge>
        {type}
      </p>
      <ReservationRocket rocketId={id} />
    </div>
  </div>
);

Rocket.propTypes = { name: PropTypes.string.isRequired, type: PropTypes.string.isRequired };
Rocket.propTypes = { id: PropTypes.number.isRequired, image: PropTypes.string.isRequired };
export default Rocket;
