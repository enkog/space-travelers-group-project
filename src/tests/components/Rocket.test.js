import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Rocket from '../../components/Rocket';

describe('Rocket component', () => {
  test('renders Rocket component', () => {
    render(
      <Provider store={store}>
        <Rocket key="0" id="0" name="test" type="Test test" image="test" reserved />
      </Provider>,
    );
    screen.getByText('test');
    expect(screen.getByText('Test test')).toBeInTheDocument();
  });
});
