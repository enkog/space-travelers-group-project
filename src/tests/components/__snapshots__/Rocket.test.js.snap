// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component snapshot test 1`] = `
<div
  className="rocket-card d-flex"
>
  <img
    alt="Rocket"
    className="rocket-image"
    src="test"
  />
  <div>
    <h1
      className="rocket-name"
    >
      test
    </h1>
    <p
      className="rocket-type"
    >
      <span
        className="tag badge bg-primary"
      >
        Reserved
      </span>
      Test test
    </p>
    <input
      className="cancel-rocket"
      onClick={[Function]}
      type="button"
      value="Cancel Rocket"
    />
  </div>
</div>
`;
